lib/watir/locators/element/locator.rb

```diff
@@ -196,13 +191,23 @@ def label_from_text
         end
 
         def matches_selector?(element, selector)
-          selector.all? do |how, what|
+          matches = selector.all? do |how, what|
             if how == :tag_name && what.is_a?(String)
               element_validator.validate(element, {tag_name: what})
             else
               what === fetch_value(element, how)
             end
           end
+
+          if selector[:text]
+            text_content = Watir::Element.new(@query_scope, element: element).send(:execute_js, :getTextContent, element).strip
+            text_content_matches = selector[:text] === text_content
+            unless matches == text_content_matches
+              Watir.logger.deprecate(':text locator with RegExp values to find elements based on only visible text', ":visible_text")
```

Comment:
What if we change this to:
```
Watir.logger.deprecate(":text locator with RegExp: #{@selector[:text].inspect} matched an element with hidden text", ":visible_text")
```

Reply:
That seems reasonable. I can update the pull request tonight if you haven't done so by then.

Reply:
@titusfortner , message has been updated.

Reply:
You have merge powers, right? Shows you're an owner, just making sure.